Auth logic

00rigin opened the pull request:
* 로그인 인증
* 인터셉터 사용하여 auth에 따라 접근 가능 구역 설정

00rigin committed:
로그인 인증

00rigin committed:
로그인 인증 / 인터셉터 사용하여 auth에 따른 접근 가능 구역 설정

sym0417 commented on backend/src/main/java/com/midasit/midascafe/security/LoginInterceptor.java:
```diff
@@ -0,0 +1,77 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
```
원래는 filter, interceptor 등등 보안처리의 모든 부분이 Spring security라는 스프링 하위 프레임워크에 다 포함되는 개념이라, 추후에 공부해놓거나 알아놓으면 좋습니다ㅎㅎ

sym0417 commented on backend/src/main/java/com/midasit/midascafe/security/LoginInterceptor.java:
```diff
@@ -0,0 +1,77 @@
+import java.util.Arrays;
+
+public class LoginInterceptor implements HandlerInterceptor {
+
```
https://techblog.woowahan.com/2527/
이런 느낌으로 리스트나 변수들을 깔끔하게 관리하는 방법도 좋은 방법이라고 생각됩니다ㅎㅎ

ymseoooooo commented on backend/src/main/java/com/midasit/midascafe/security/WebConfig.java:
```diff
@@ -0,0 +1,15 @@
+public class WebConfig implements WebMvcConfigurer {
+    @Override
+    public void addInterceptors(InterceptorRegistry registry){
+        registry.addInterceptor(new LoginInterceptor())
```
지금 interceptor에서 모든 권한들을 처리 중인데,
권한 별로 interceptor를 분리하면 
`userPermissioned` `nonePermissioned` 이렇게 리스트로 관리 안해도 될 것 같습니다!

00rigin committed:
*인터셉터 분리
*쿠기 기반 사용자 auth 확인으로 변경
*로그인한 사용자의 주문 내역 확인 기능
*관리자의 특정 사용자 주문 확인 기능
*관리자의 wait 상태 주문 확인 기능

yhj0122 committed:
* 사용하지 않는 변수 제거
* 조회 효율 향상

yhj0122 committed:
* 불필요 함수 제거

00rigin committed:
Merge pull request #10 from 00rigin/advanced_function

*인터셉터 분리
*쿠기 기반 사용자 auth 확인으로 변경
*로그인한 사용자의 주문 내역 확인 기능
*관리자의 특정 사용자 주문 확인 기능
*관리자의 wait 상태 주문 확인 기능
